### PR TITLE
Have curation platform suports priority

### PR DESCRIPTION
**Pull request by zhx828**
1. Update priority when moving treatments
2. Add priority when adding new treatment
3. After approving edits on treatment, update priorities

### CODE REVIEW

File: web/yo/app/scripts/controllers/gene.js [hunk 1]
```diff
@@ -2892,6 +2907,88 @@ angular.module('oncokbApp')
...
+                    // no need to trigger update.
+                    if (priorities[indexUUid] && priorities[moveIndexUUid]) {
+                        postData[indexUUid] = priorities[indexUUid];
+                        postData[moveIndexUUid] = priorities[moveIndexUUid];
+                    }
+                } else {
+                    postData = priorities;
+                }
+
+                if (Object.keys(postData).length > 0) {
```
Comment (JiaoJiao123) [inline web/yo/app/scripts/controllers/gene.js]: needs to do isDesiredGene and internal network check

File: core/src/main/java/org/mskcc/cbio/oncokb/bo/GenericBo.java [hunk 2]
```diff
@@ -19,8 +19,6 @@
 
     void saveOrUpdate(T t);
 
-    void saveOrUpdateAll(List<T> ts);
```
Comment (JiaoJiao123) [inline core/src/main/java/org/mskcc/cbio/oncokb/bo/GenericBo.java]: can we do saveOrUpdateAll on Set like this? So we can update a batch
void saveOrUpdateAll(Set<T> ts);

Comment (zhx828) [reply core/src/main/java/org/mskcc/cbio/oncokb/bo/GenericBo.java]: As I mentioned in the commit, this method is no longer supported.

File: web/yo/app/scripts/controllers/gene.js (hunk 1 above)
Comment (zhx828) [reply web/yo/app/scripts/controllers/gene.js]: I can add isDesiredGene, but internal check should be inside databaseconnector as wel discussed before.

File: web/yo/app/scripts/controllers/gene.js [hunk 3]
```diff
@@ -1758,6 +1763,7 @@ angular.module('oncokbApp')
                     treatment.name_review.delete('added');
                     ReviewResource.accepted = _.union(ReviewResource.accepted, [treatment.name_uuid.getText(), treatment.level_uuid.getText(), treatment.indication_uuid.getText(), treatment.description_uuid.getText()]);
                     clearReview([treatment.name_review, treatment.level_review, treatment.indication_review, treatment.description_review]);
+                    updatePriority(ti.treatments);
```
[the remark this answers is not in the document]
Comment (JiaoJiao123) [reply web/yo/app/scripts/controllers/gene.js]: just in case you didn't see it, 'firstLayer' is a parameter already passed into acceptSectionItems().

**Pull request merged**